duckduckgo/Android
# Keep HTTP auth credentials when site is fireproof(ed)

Task/Issue URL: https://app.asana.com/0/414730916066338/1199538898514852/f
Tech Design URL: https://app.asana.com/0/0/1199538898514868/f
CC: 

**Description**:
This task fixes the issue that makes the HTTP auth credentials being removed upon firing even when the site is fireproofed.

As the `httpauth` table db where the credentials are stored is never read by Android, we can manually read/write/delete at will without interfering with Android.
We wrap the DB with Room just as a convenience.


**Steps to test this PR**:
Verify fireproofing works with migration
* install the app from develop branch
* visit https://httpbin.org/basic-auth/test/1111
* enter user = test, password = 1111
* fire
* visit https://httpbin.org/basic-auth/test/1111 
* verify the Auth dialog is shown again (credentials have been deleted)
* enter user = test, password = 1111
* upgrade the app from this branch
* launcher app and visit https://httpbin.org/basic-auth/test/1111 
* fire
* visit https://httpbin.org/basic-auth/test/1111 
* verify the Auth dialog is shown again (credentials have been deleted)
* enter user = test, password = 1111
* fireproof the site
* fire
* visit https://httpbin.org/basic-auth/test/1111 
* verify auth dialog is no longer shown and user is already authenticated

Verify fireproofing works with clean build
* Clean install the app from this branch
* visit https://httpbin.org/basic-auth/test/1111 
* enter user = test, password = 1111
* fire
* visit https://httpbin.org/basic-auth/test/1111 
* verify the Auth dialog is shown again (credentials have been deleted)
* enter user = test, password = 1111
* fireproof the site
* fire
* visit https://httpbin.org/basic-auth/test/1111 
* verify auth dialog is no longer shown and user is already authenticated

Verify credentials can be updated
* install the app from this branch
* visit https://httpbin.org/basic-auth/test/1111 
* enter user = test, password = 1111
* close and re-open the app
* visit https://httpbin.org/basic-auth/test/1111 
* verify user is logged in
* visit https://httpbin.org/basic-auth/test/2222
* verify auth dialog appears
* enter user = test, password = 2222
* verify user is logged in
* visit https://httpbin.org/basic-auth/test/1111 
* verify auth dialog appears

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
